lib/oban/job.ex

```diff
@@ -36,6 +36,17 @@ defmodule Oban.Job do
           | {:period, unique_period()}
           | {:states, [unique_state()]}
 
+  @type replace_option :: [
+          :args
+          | :max_attempts
+          | :meta
+          | :priority
+          | :queue
+          | :scheduled_at
+          | :tags
+          | :worker
```

Comment:
Technically the `worker` and the `queue` are part of the unique constraints and viable viable options to `Job.new`. it seems strange to change them on a conflict, but I can't think of a reason why _not_ to do it.